client/client.go

```diff
@@ -154,39 +136,30 @@ func (c *Client) DialDefault() (*Conn, error) {
 	return c.DefaultRemote.Dial(c)
 }
 
-// registerSKI associates the SKI of a public key with a particular keyserver.
-func (c *Client) registerSKI(server string, ski gokeyless.SKI) error {
-	c.m.Lock()
-	defer c.m.Unlock()
+// getRemote obtains the Remote for the given server name, so that it can be
+// associated with the PrivateKey
+func (c *Client) getRemote(server string) (Remote, error) {
 	// empty server means always associate ski with DefaultRemote
 	if server == "" {
-		log.Debugf("registering key @default_remote with SKI: %02x", ski)
-		c.remotes[ski] = c.DefaultRemote
-		return nil
+		return c.DefaultRemote, nil
 	}
 
-	log.Debugf("registering key @%s with SKI: %02x", server, ski)
-	cachedRemote := false
 	v, stale := c.remoteCache.Get(server)
 	if v != nil && !stale {
 		if r, ok := v.(Remote); ok {
-			c.remotes[ski] = r
-			cachedRemote = true
+			return r, nil
 		} else {
-			log.Error("failed to convert cached remote")
+			panic("failed to convert cached remote")
```

Comment:
this should not be unlikely, but simply impossible unless there is a programmer error in adding (and guru tells me there's not), so it's the kind of place where panic() is appropriate IMHO, but I know that's subjective